demisto_sdk/commands/update_release_notes/update_rn_manager.py

```diff
@@ -0,0 +1,194 @@
+                 prev_ver: Optional[str] = None, is_force: bool = False):
+        self.given_pack = user_input
+        self.changed_packs_from_git: set = set()
+        self.update_type = update_type
+        self.pre_release: bool = False if pre_release is None else pre_release
+        # update release notes to every required pack if not specified.
+        self.is_all = True if not self.given_pack else is_all
+        self.text: str = '' if text is None else text
+        self.is_force = is_force
+        self.specific_version = specific_version
+        self.id_set_path = id_set_path
+        self.prev_ver = prev_ver
+        self.packs_existing_rn: dict = {}
+        self.total_updated_packs: set = set()
+        # When a user choose a specific pack to update rn, the -g flag should not be passed
+        if self.given_pack and self.is_all:
+            raise ValueError('Please remove the -g flag when specifying only one pack.')
+
+    def manage_rn_update(self):
+        """
+            Manages the entire update release notes process.
+        """
+        print_color('Starting to update release notes.', LOG_COLORS.NATIVE)
+        # The given_pack can be both path or pack name thus, we extract the pack name from the path if needed.
+        if self.given_pack and '/' in self.given_pack:
+            self.given_pack = get_pack_name(self.given_pack)  # extract pack from path
+
+        # Find which files were changed from git
+        modified_files, added_files, _, old_format_files = self.get_git_changed_files()
+        self.changed_packs_from_git = get_pack_names_from_files(modified_files).union(
+            get_pack_names_from_files(added_files)).union(get_pack_names_from_files(old_format_files))
+        # Check whether the packs have some existing RNs already (created manually or by the command)
+        self.check_existing_rn(added_files)
+
+        self.handle_api_module_change(modified_files, added_files)
+        self.create_release_notes(modified_files, added_files, old_format_files)
+        if len(self.total_updated_packs) > 1:
+            print_color('\nSuccessfully updated the following packs:\n' + '\n'.join(self.total_updated_packs),
+                        LOG_COLORS.GREEN)
+
+    def get_git_changed_files(self) -> Tuple[set, set, set, set]:
+        """ Get the changed files from git (added, modified, old format, metadata).
+
+            :return:
+                4 sets:
+                - The filtered modified files (including the renamed files)
+                - The filtered added files
+                - The changed metadata files
+                - The modified old-format files (legacy unified python files)
+        """
+        try:
+            validate_manager = ValidateManager(skip_pack_rn_validation=True, prev_ver=self.prev_ver,
+                                               silence_init_prints=True, skip_conf_json=True, check_is_unskipped=False)
```

Comment:
might be wise to add support for the debug-git flag as well. though not mandatory.